wallet/src/de/schildbach/wallet/ui/InputParser.java

```diff
@@ -443,10 +443,10 @@ protected void cannotClassify(final String input) {
     private static final Pattern PATTERN_BITCOIN_ADDRESS = Pattern
             .compile("[" + new String(Base58.ALPHABET) + "]{20,40}");
     private static final Pattern PATTERN_DUMPED_PRIVATE_KEY_UNCOMPRESSED = Pattern
-            .compile((Constants.NETWORK_PARAMETERS.getId().equals(NetworkParameters.ID_MAINNET) ? "X" : "9") + "["
+            .compile((Constants.NETWORK_PARAMETERS.getId().equals(NetworkParameters.ID_MAINNET) ? "7" : "9") + "["
                     + new String(Base58.ALPHABET) + "]{50}");
     private static final Pattern PATTERN_DUMPED_PRIVATE_KEY_COMPRESSED = Pattern
-            .compile((Constants.NETWORK_PARAMETERS.getId().equals(NetworkParameters.ID_MAINNET) ? "7" : "c") + "["
+            .compile((Constants.NETWORK_PARAMETERS.getId().equals(NetworkParameters.ID_MAINNET) ? "X" : "c") + "["
```

Comment:
in a previous commit on this `features-core-0.18` branch the prefix letters 7 and X were reversed which wasn't detected until some tests failed.